src/main/java/com/datasonnet/document/JavaObjectDocument.java

```diff
@@ -0,0 +1,24 @@
+package com.datasonnet.document;
+
+public class JavaObjectDocument implements Document<Object> {
+
+    private Object contents;
+
+    public JavaObjectDocument(Object contents) {
+        this.contents = contents;
+    }
+
+    @Override
+    public Object getContents() {
+        return contents;
+    }
+
+    @Override
+    public String getMimeType() {
+        return "application/java";
+    }
+
+    public boolean equals(JavaObjectDocument document) {
+        return (contents.equals(document.getContents()) && "application/java".equalsIgnoreCase(document.getMimeType()));
```

Comment:
TBH I forgot why I added it - I think it could be a good idea to have it for future performance improvements, when we add some caching mechanisms. But I can remove or comment it out for now.

Reply:
Deleting is good, we can always assess and add if we need it later